scripts/hgsc_subtypes_tybalt.py

```diff
@@ -0,0 +1,308 @@
+rnaseq_df = pd.read_table(rnaseq_file, index_col=0)
+rnaseq_df.shape
+
+
+# In[6]:
+
+ov_file = os.path.join('data', 'ov_subtype_info.tsv')
+ov_df = pd.read_table(ov_file, index_col=0)
+ov_df.head(2)
+
+
+# In[7]:
+
+encoded_file = os.path.join('data', "encoded_rnaseq_onehidden_warmup_batchnorm.tsv")
+encoded_df = pd.read_table(encoded_file, index_col=0)
+encoded_df.shape
+
+
+# In[8]:
+
+# Subset and merge the HGSC subtype info with the latent space feature activations
+ov_samples = list(set(encoded_df.index) & (set(ov_df.index)))
+
+ov_encoded = encoded_df.loc[ov_samples, ]
+ov_encoded_subtype = pd.merge(ov_df.loc[:, ['SUBTYPE', 'SILHOUETTE WIDTH']], ov_encoded,
+                              how='right', left_index=True, right_index=True)
+ov_encoded_subtype = ov_encoded_subtype.assign(subtype_color = ov_encoded_subtype['SUBTYPE'])
+
+ov_subtype_color_dict = {'Differentiated': 'purple',
+                         'Immunoreactive': 'green',
+                         'Mesenchymal': 'blue',
+                         'Proliferative': 'red'}
+ov_encoded_subtype = ov_encoded_subtype.replace({'subtype_color': ov_subtype_color_dict})
+
+print(ov_encoded_subtype.shape)
+ov_encoded_subtype.head(2)
+
+
+# In[9]:
+
+# Get the HGSC mean feature activation
+ov_mean_subtypes = ov_encoded_subtype.groupby('SUBTYPE').mean()
+ov_mean_subtypes
+
+
+# ## HGSC Subtype Math
+# 
+# Because of the relationship observed in the consistent clustering solutions, perform the following subtractions
+# 
+# 1. Immunoreactive - Mesenchymal
+# 2. Differentiated - Proliferative
+# 
+# The goal is to observe the features with the largest difference between the aformentioned comparisons. The differences should be in absolute directions
+
+# In[10]:
+
+mes_mean_vector = ov_mean_subtypes.loc['Mesenchymal', [str(x) for x in range(1, 101)]]
+imm_mean_vector = ov_mean_subtypes.loc['Immunoreactive', [str(x) for x in range(1, 101)]]
+pro_mean_vector = ov_mean_subtypes.loc['Proliferative', [str(x) for x in range(1, 101)]]
+dif_mean_vector = ov_mean_subtypes.loc['Differentiated', [str(x) for x in range(1, 101)]]
+
+
+# In[11]:
+
+high_immuno = (imm_mean_vector - mes_mean_vector).sort_values(ascending=False).head(2)
+high_mesenc = (imm_mean_vector - mes_mean_vector).sort_values(ascending=False).tail(2)
+
+print("Features with large differences: Immuno high, Mesenchymal low")
+print(high_immuno)
+print("Features with large differences: Mesenchymal high, Immuno low")
+print(high_mesenc)
+
+
+# In[12]:
+
+# Select to visualize encoding 56 because it has high immuno and low everything else
+ov_mean_subtypes.loc[:, ['87', '77', '56']]
+
+
+# In[13]:
+
+# Obtain the decoder weights
+weights = []
+for l in decoder.layers:
+    weights.append(l.get_weights())
+    
+weight_layer = pd.DataFrame(weights[1][0], columns=rnaseq_df.columns)
+weight_layer.head(2)
+
+
+# In[14]:
+
+# Get the high weight genes describing the immunoreactive subtype
```

Comment:
Since you will be doing these operations (i.e., identifying high weight genes and writing them to file) for multiple nodes, I think it would be helpful to write some functions to this and it would shorten things quite a bit. You might also consider identifying high weight genes for _all_ nodes instead of doing so selectively?

Reply:
> I think it would be helpful to write some functions to this and it would shorten things quite a bit.

It's probable - I will look into this

> You might also consider identifying high weight genes for all nodes instead of doing so selectively?

I think it would be preferable to save the entire weight matrix (rather than extracting it each time from the model). Extracting genes from this matrix is quick and can be customized according to specific needs. The next PR will output a gene weight matrix, so perhaps we leave this change until then.

Reply:
once merged it will address #64

Reply:
Okay, saving the entire weight matrix sounds good to me!